### PR TITLE
imx9_flexcan: use small lock in arch/arm/src/imx9/imx9_flexcan.c

### PR DESCRIPTION
**Pull request by hujun260**
## Summary
imx9_flexcan: use small lock in arch/arm/src/imx9/imx9_flexcan.c
reason:
We would like to replace the big lock with a small lock.

## Impact
imx9_flexcan

## Testing
ci

### CODE REVIEW

**Comment by nuttxpr** [comment]
[**\[Experimental Bot, please feedback here\]**](https://github.com/search?q=repo%3Aapache%2Fnuttx+13552&type=issues)



No, this PR does not fully meet the NuttX requirements.  While it provides a basic summary and identifies the impacted area, it lacks crucial details.

Here's a breakdown of what's missing:

* **Insufficient Summary:**  While the reason for the change is stated, the summary lacks specifics.  It doesn't describe *how* the change works (the exact mechanism of replacing the lock) or *what functional part* of `imx9_flexcan.c` is modified.  Mentioning functions, data structures, or specific code sections affected would be helpful.  Linking a related NuttX issue is also highly recommended.

* **Incomplete Impact Assessment:** Simply stating "imx9_flexcan" isn't sufficient.  The PR needs to address all the listed impact categories (user impact, build impact, hardware impact, documentation, security, compatibility).  For each, explicitly state "NO" or "YES" and provide a description if "YES."  For example, even if the answer is "NO" for most, the PR should state "Impact on hardware: NO" for clarity.  Since this change deals with locking, there's a potential impact on performance and real-time behavior, which needs to be addressed.

* **Inadequate Testing:** "ci" is not enough.  The PR must include:
    * Specific build host details (OS, CPU architecture, compiler version).
    * Specific target details (architecture, board, configuration).
    * **Actual testing logs** before and after the change demonstrating the intended effect.  Just stating "ci" doesn't show *what* was tested or *how* the change improved things.  Ideally, logs should show improved performance or demonstrate the resolution of a specific issue.


**In short, the PR needs to be significantly more detailed to meet the NuttX requirements.**  It needs to explain the change thoroughly, analyze its impact comprehensively, and provide concrete evidence of testing.

**Review by jerpelea — APPROVED** [review]

File: arch/arm/src/imx9/imx9_flexcan.c [hunk 1]
```diff
@@ -811,7 +812,7 @@ static int imx9_txpoll(struct net_driver_s *dev)
    * the field d_len is set to a value > 0.
    */
 
-  flags = spin_lock_irqsave(NULL);
+  flags = spin_lock_irqsave(&priv->lock);
```
Comment (anchao) [inline arch/arm/src/imx9/imx9_flexcan.c:815]: sched_lock

Comment (hujun260) [reply arch/arm/src/imx9/imx9_flexcan.c:815]: But I didn't find any calls to sem_post in the protected code.

Comment (anchao) [reply arch/arm/src/imx9/imx9_flexcan.c:815]: https://github.com/apache/nuttx/blob/master/arch/arm/src/imx9/imx9_flexcan.c#L645

Comment (hujun260) [reply arch/arm/src/imx9/imx9_flexcan.c:815]: done

**Review by xiaoxiang781216 — APPROVED** [review]

**Pull request merged**